Integrations/integration-RTIR.yml

```diff
@@ -585,16 +684,54 @@ script:
             demisto.setLastRun({'ticket_id': max_ticket_id})
         demisto.incidents(incidents)
 
+
     ''' EXECUTION CODE '''
 
-    tracker = rt.Rt(BASE_URL, USERNAME, PASSWORD, verify_cert=USE_SSL)
-    login = tracker.login()
-    if login is False:
-        return_error('Failed to login')
 
-    LOG('command is %s' % (demisto.command(), ))
+    def http_request(method, suffix_url, params=None, data=None, files=None):
+        #Returns the http request
 
+        url = BASE_URL + suffix_url
+        response = SESSION.request(method, url, data=data, params=params, files=files)
+
+        # handle request failure
+        if response.status_code not in {200}:
+            message = parse_error_response(response)
+            return_error('Error in API call with status code {}\n{}'.format(response.status_code, message))
+
+        return response
+
+
+    def parse_error_response(response):
+        try:
+            res = response.json()
+            msg = res.get('message')
+            if res.get('details') is not None and res.get('details')[0].get('message') is not None:
```

Comment:
no need to check `is not None`, but `res.get('details)` is enough, same for the second part of the condition